client/my-sites/sidebar/sidebar.jsx

```diff
@@ -681,10 +678,16 @@ export const MySitesSidebar = React.createClass( {
 	},
 
 	render: function() {
-		var publish = !! this.publish(),
-			appearance = ( !! this.themes() || !! this.menus() ),
-			configuration = ( !! this.sharing() || !! this.users() || !! this.siteSettings() || !! this.plugins() || !! this.upgrades() ),
-			vip = !! this.vip();
+		const showPublishMenu = !! this.renderPublishMenu();
+		const showAppearanceMenu = !! this.renderThemesOption() || !! this.renderMenusOption();
+		const showConfigurationMenu = (
+			!! this.renderSharingOption() ||
+			!! this.renderUsersOption() ||
+			!! this.renderSettingsOption() ||
+			!! this.renderPluginsOption() ||
+			!! this.renderUpgradesOption()
+		);
```

Comment:
> I didn't want to bring new checking methods in to the scope of this PR but I am planning to add some

it may be necessary to refactor if we are going to touch it at all 😢 

> I usually prefer to do the checking separately first and just not even attempt to call this function 

sounds very good

> the responsibility should come from the caller, not the component.

sounds even better